nomad/structs/csi.go

```diff
@@ -578,6 +578,11 @@ func (v *CSIVolume) Validate() error {
 	if v.AttachmentMode == "" {
 		errs = append(errs, "missing attachment mode")
 	}
+	if v.AttachmentMode == CSIVolumeAttachmentModeBlockDevice {
+		if v.MountOptions != nil {
```

Comment:
Do we want to be forgiving if it's set but empty?  Being a bit more relaxed here may ease some templating/job submitting frameworks?

```suggestion
		if len(v.MountOptions) != 0 {
```

Reply:
`MountOptions` is a pointer, so we'll have to check its contents. But yeah, that seems like a good idea.

Reply:
Done below.